lib/hal_client/retryinator.rb

```diff
@@ -0,0 +1,45 @@
+require 'hal_client/null_logger'
+
+class HalClient
+  class Retryinator
+
+    attr_reader :max_tries, :interval, :logger
+
+    DEFAULT_MAX_TRIES = 3
+    DEFAULT_INTERVAL = 1
+
+    def initialize(options={})
+      @max_tries = options.fetch(:max_tries, DEFAULT_MAX_TRIES)
+      @interval = options.fetch(:interval, DEFAULT_INTERVAL)
+      @logger = options.fetch(:logger, HalClient::NullLogger.new)
+    end
+
+    def call(&block)
+      current_try = 1
+
+      loop do
+        begin
+          result = yield block
+
+          if server_error?(result.code)
```

Comment:
`result.status.server_error?`

Reply:
that was what we had originally, but the tests for older versions of http that are set up on this project broke due to that method not being present on the status object